external/wakaama/examples/client/lwm2mclient.c

```diff
@@ -1185,7 +1187,7 @@ pthread_addr_t client_main(void)
 
 		if (NULL == pskBuffer) {
 			fprintf(stderr, "Failed to create PSK binary buffer\r\n");
-			return -1;
+			return NULL;
```

Comment:
Not sure I understand correctly, you return NULL at similar error conditions. Yet, at the end, when everything is successfully done, we return 0. Is this something you would want to revise?